sql/threadpool_unix.cc

```diff
@@ -1328,6 +1328,13 @@ bool Thread_pool_connection_handler::add_connection(Channel_info *channel_info)
   DBUG_ENTER("Thread_pool_connection_handler::add_connection");
 
   THD* thd= channel_info->create_thd();
+
+  if (unlikely(!thd))
+  {
+    channel_info->send_error_and_close_channel(ER_OUT_OF_RESOURCES, 0, false);
+    DBUG_RETURN(true);
```

Comment:
No - if true is returned caller deletes it. But this shows that the next exit from the function, if alloc_connection calls returns NULL, is wrong. Fixing.